tests/components/intellifire/test_config_flow.py

```diff
@@ -1,55 +1,205 @@
 """Test the IntelliFire config flow."""
-from unittest.mock import AsyncMock, MagicMock
+from unittest.mock import AsyncMock, MagicMock, patch
 
 from homeassistant import config_entries
+from homeassistant.components.intellifire.config_flow import MANUAL_ENTRY_STRING
 from homeassistant.components.intellifire.const import DOMAIN
+from homeassistant.const import CONF_HOST
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
 
+from tests.common import MockConfigEntry
 
-async def test_form(
+
+async def test_no_discovery(
     hass: HomeAssistant,
     mock_setup_entry: AsyncMock,
     mock_intellifire_config_flow: MagicMock,
 ) -> None:
-    """Test we get the form."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
+    """Test we should get the manual discovery form - because no discovered fireplaces."""
+    with patch(
+        "homeassistant.components.intellifire.config_flow.AsyncUDPFireplaceFinder.search_fireplace",
+        return_value=[],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
     assert result["type"] == RESULT_TYPE_FORM
-    assert result["errors"] is None
+    assert result["errors"] == {}
+    assert result["step_id"] == "manual_device_entry"
 
     result2 = await hass.config_entries.flow.async_configure(
         result["flow_id"],
         {
-            "host": "1.1.1.1",
+            CONF_HOST: "1.1.1.1",
         },
     )
     await hass.async_block_till_done()
-
     assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
-    assert result2["title"] == "Fireplace"
-    assert result2["data"] == {"host": "1.1.1.1"}
+    assert result2["title"] == "Fireplace 12345"
+    assert result2["data"] == {CONF_HOST: "1.1.1.1"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_single_discovery(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test single firedplace UDP discovery."""
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
```

Comment:
There are few more patch targets that could be cleaned up to only patch where it is needed.